tests/sles4sap/sap_deployment_automation_framework/publiccloud_lib_compatibility_layer.pm

```diff
@@ -0,0 +1,87 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Prepares compatibility layer for using `lib/publiccloud/*` library with SDAF deployment
+
+use parent 'sles4sap::sap_deployment_automation_framework::basetest';
+
+use warnings;
+use strict;
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use sles4sap::console_redirection qw(connect_target_to_serial disconnect_target_from_serial);
+use sles4sap::sap_deployment_automation_framework::inventory_tools qw(read_inventory_file sdaf_create_instances);
+use sles4sap::sap_deployment_automation_framework::naming_conventions
+  qw(get_sdaf_inventory_path convert_region_to_short get_sdaf_config_path get_workload_vnet_code);
+
+sub run {
+    my ($self, $run_args) = @_;
+    select_serial_terminal;
+    my $workload_vnet_code = get_workload_vnet_code();
+    my $sap_sid = get_required_var('SAP_SID');
+    my $sdaf_region_short = convert_region_to_short(get_required_var('PUBLIC_CLOUD_REGION'));
+    my $sdaf_env_code = get_required_var('SDAF_ENV_CODE');
+
+    my $inventory_path = get_sdaf_inventory_path(
+        env_code => $sdaf_env_code,
+        sdaf_region_code => $sdaf_region_short,
+        vnet_code => $workload_vnet_code,
+        sap_sid => $sap_sid
+    );
+
+    my $sut_ssh_private_key = get_sdaf_config_path(
+        deployment_type => 'sap_system',
+        env_code => $sdaf_env_code,
+        sdaf_region_code => $sdaf_region_short,
+        vnet_code => $workload_vnet_code,
+        sap_sid => $sap_sid) . '/sshkey';
+
+    # Redirect serial to deployer VM. Deployer VM takes same role as worker VM.
+    # Normally PC test runs under root user
+    connect_target_to_serial(switch_root => 'yes');
+
+    # PC lib uses 'bernhard' user by default - change global variable to override this default value
+    $testapi::username = "root";
+
+    my $inventory_data = read_inventory_file($inventory_path);
```

Comment:
why needs to be done after `connect_target_to_serial`? Is it because inventory.yaml is on the remote machine?

Reply:
Yes. We need to login to the machine and read the file.